### PR TITLE
Add DSR file and ajust new path of platform

### PR DESCRIPTION
**Pull request by Isengo1989**
Add input to pass plugin name and url to the workflow. This enables to create separate JSON files to be created for:

- SwagCommercial
- SwagGuidedShopping
- ....

Open question: Do we want to split up SwagCommercial in the future? IMO, it is more clear what is a feature of Evolve, Beyond etc.

image

@elkmod @bojanrajh @Christian-Rades

### CODE REVIEW

File: .github/workflows/generate_schema.yml [hunk 1]
```diff
@@ -70,29 +88,41 @@ jobs:
...
       run: |
         git checkout $SWTAG
-        
-    - name: checkout commercial plugin
+
+    - name: checkout ${{ inputs.plugin_name }} plugin
       env:
         DEPLOY_USER: ${{ secrets.DEPLOY_USER }}
         DEPLOY_TOKEN: ${{ secrets.DEPLOY_TOKEN }}
+        DIGITALSALESROOM_DEPLOY_TOKEN: ${{ secrets.DIGITALSALESROOM_DEPLOY_TOKEN }}
```
Comment (bojanrajh) [inline .github/workflows/generate_schema.yml]: Seems like we can reuse the `DEPLOY_USER` and `DEPLOY_TOKEN` instead, add them to [`with`](https://github.com/shopware/admin-api-reference/blob/main/.github/workflows/update-api.yml#L12)?

File: .github/workflows/generate_schema.yml [hunk 2]
```diff
@@ -70,29 +88,41 @@ jobs:
...
+        DIGITALSALESROOM_DEPLOY_USER: ${{ secrets.DIGITALSALESROOM_DEPLOY_USER }}
+        DAILY_CO_API_KEY: ${{ secrets.DAILY_CO_API_KEY }}
       run: |
-        git clone https://${DEPLOY_USER}:${DEPLOY_TOKEN}@gitlab.shopware.com/shopware/6/product/commercial.git custom/plugins/SwagCommercial
-        
+        if [ "${{ inputs.plugin_name }}" = "SwagGuidedShopping" ]; then
+          git clone --single-branch --branch 6.5-compatibility https://${DIGITALSALESROOM_DEPLOY_USER}:${DIGITALSALESROOM_DEPLOY_TOKEN}@${{ inputs.plugin_url }} custom/plugins/${{ inputs.plugin_name }}
```
Comment (bojanrajh) [inline .github/workflows/generate_schema.yml]: This (`--single-branch --branch 6.5-compatibility`) could be added as `input.GIT_CLONE_OPTIONS`.

File: .github/workflows/generate_schema.yml [hunk 3]
```diff
@@ -41,22 +49,32 @@ jobs:
...
+          repository: shopware/shopware
           ref: trunk
           fetch-depth: 0
 
     - name: configure job
       run: |
         if [ "${{ inputs.api_type }}" = "storeapi" ]; then
-          echo "SCHEMA_FLAGS=-s openapi3 --store-api storeapi.json" >> $GITHUB_ENV
-          echo "SCHEMA_FILE=storeapi.json" >> $GITHUB_ENV
+          if [ "${{ inputs.plugin_name }}" ]; then
```
Comment (bojanrajh) [inline .github/workflows/generate_schema.yml]: If I understand correctly, we'll end up with 4 schemas (base + 2 for each plugin)?
 - adminapi
 - storeapi
 - DSL-adminapi
 - DSL-storeapi

Wouldn't it be easier to add `SCHEMA_FLAGS` and `SCHEMA_FILE` to the `input`?

**Review by bojanrajh — CHANGES_REQUESTED** [review]

File: .github/workflows/generate_schema.yml (hunk 1 above)
Comment (Isengo1989) [reply .github/workflows/generate_schema.yml]: We always get one user + token per plugin repo (regulation by devops team). I would rather rename them to the specific use case later on.

COMMERCIAL_DEPLOY_TOKEN etc.

As for the assignment, not sure if this makes a difference. The scopes  in github do irritate me sometimes of the env / vars etc. :smile:

File: .github/workflows/generate_schema.yml (hunk 3 above)
Comment (Isengo1989) [reply .github/workflows/generate_schema.yml]: It would be 6 - depending on that, we do want to split up the commercial (see initial question above).

**Comment by elkmod** [comment]
~~I think it would make sense to separate the endpoints by "Features" rather than versions. For the respective features we could still mark the corresponding version somehow 🤔~~

EDIT after Weekly: Let's organize the endpoints by module/extension name

* Core
* Digital Sales Rooms
* Commercial
  * Quote Management
  * Customer Specific Pricing 
  * etc.

**Pull request merged**